test-case/check-audio-equalizer.sh

```diff
@@ -64,30 +66,41 @@ func_test_eq()
 }
 
 # this function performs IIR/FIR filter test
-# param1 must be must be iir or fir
+# param1 must be must be component name
 func_test_filter()
 {
     local testfilter=$1
     dlogi "Get amixer control id for $testfilter"
-    # TODO: Need to match alsa control id with the filter in the pipeline,
-    #       currently the test discards EQ pipelines except first one.
-    Filterid=$(amixer -D hw:"$sofcard" controls | sed -n -e "/eq${testfilter}/I "'s/numid=\([0-9]*\),.*/\1/p' | head -1)
+    Filterid=`$my_dir/../tools/topo_effect_kcontrols.py $tplg $testfilter`
```

Comment:
SC2006

There are some other shellcheck warnings, please take a look and fix all the easy ones.